adapter/internal/operator/controllers/dp/jwtissuer_controller.go

```diff
@@ -0,0 +1,250 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package dp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/wso2/apk/adapter/internal/discovery/xds"
+	"github.com/wso2/apk/adapter/internal/loggers"
+	"github.com/wso2/apk/adapter/internal/operator/apis/dp/v1alpha1"
+	dpv1alpha1 "github.com/wso2/apk/adapter/internal/operator/apis/dp/v1alpha1"
+	"github.com/wso2/apk/adapter/internal/operator/constants"
+	"github.com/wso2/apk/adapter/internal/operator/utils"
+	"github.com/wso2/apk/adapter/pkg/discovery/api/wso2/discovery/subscription"
+	"github.com/wso2/apk/adapter/pkg/logging"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8client "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	jwtIssuerIndex       = "jwtIssuerIndex"
+	secretJWTIssuerIndex = "secretJWTIssuerIndex"
+	configmapIssuerIndex = "configmapIssuerIndex"
+)
+
+// JWTIssuerReconciler reconciles a JWTIssuer object
+type JWTIssuerReconciler struct {
+	client client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=dp.wso2.com,resources=jwtissuers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the JWTIssuer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *JWTIssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+	var err error
+
+	loggers.LoggerAPKOperator.Debugf("Reconciling jwtIssuer: %v", req.NamespacedName.String())
+
+	jwtKey := req.NamespacedName
+	var jwtIssuerList = new(dpv1alpha1.JWTIssuerList)
+	if err := r.client.List(ctx, jwtIssuerList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get jwtIssuer %s/%s",
+			jwtKey.Namespace, jwtKey.Name)
```

Comment:
jwtKey.Namespace, jwtKey.Name these are not string type right, doesn't it give issues due to `%s/%s`